examples/step-68/step-68.cc

```diff
@@ -0,0 +1,810 @@
+        &particle_handler));
+
+    background_triangulation.signals.post_distributed_repartition.connect(
+      std::bind(
+        &Particles::ParticleHandler<dim>::register_load_callback_function,
+        &particle_handler,
+        false));
+
+    // This initializes the background triangulation where the particles are
+    // living and the number of properties of the particles.
+    particle_handler.initialize(background_triangulation, mapping, 1 + dim);
+
+    // We create a particle triangulation which is solely used to generate
+    // the points which will be used to insert the particles. This
+    // triangulation is an hyper_shell which is off-set from the
+    // center of the simulation domain. This will be used to generate a
+    // disk filled with particles which will allow an easy monitoring
+    // of the motion due to the vortex.
+    Point<dim> center;
+    center[0] = 0.5;
+    center[1] = 0.75;
+    if (dim == 3)
+      {
+        center[2] = 0.5;
+      }
+
+    const double outer_radius = 0.15;
+    const double inner_radius = 0.01;
+
+    parallel::distributed::Triangulation<dim> particle_triangulation(
+      MPI_COMM_WORLD);
+
+    GridGenerator::hyper_shell(
+      particle_triangulation, center, inner_radius, outer_radius, 6);
+    particle_triangulation.refine_global(par.particle_insertion_refinement);
+
+    // We generate the necessary bounding boxes for the particles generator
+    // These bounding boxes are required to quickly identify in which
+    // processors and which cell the inserted particle lies.
+    const auto my_bounding_box = GridTools::compute_mesh_predicate_bounding_box(
+      background_triangulation, IteratorFilters::LocallyOwnedCell());
+    const auto global_bounding_boxes =
+      Utilities::MPI::all_gather(MPI_COMM_WORLD, my_bounding_box);
+
+    // The quadrature points particle generator generates particles only
+    // on locally owned active cells. We therefore count how many of those
+    // are present in the triangulation, as this will be required to
+    // initialize the properties.
+    unsigned int n_locally_owned_cells = 0;
+    for (const auto &cell : particle_triangulation.active_cell_iterators())
+      if (cell->is_locally_owned())
+        n_locally_owned_cells++;
+
+    // We generate an empty vector of properties. We will attribute the
+    // properties to the particles once they are generated.
+    std::vector<std::vector<double>> properties(n_locally_owned_cells,
+                                                std::vector<double>(dim + 1,
+                                                                    0.));
+
+    // We generate the particles at the position of a single
+    // points quadrature. Consequently, one particle will be generated
+    // at the centroid of each cell.
+    Particles::Generators::quadrature_points(particle_triangulation,
+                                             QGauss<dim>(1),
```

Comment:
This might be easier to understand:
```suggestion
                                             QMidpoint<dim>(),
```